shared/src/commonMain/kotlin/co/touchlab/kampkit/models/DataState.kt

```diff
@@ -0,0 +1,8 @@
+package co.touchlab.kampkit.models
+
+sealed class DataState<out T> {
+    data class Success<T>(val data: T) : DataState<T>()
+    data class Error(val exception: String) : DataState<Nothing>()
```

Comment:
Something to watch out for: this `exception` string is being passed directly to the UI and displayed. In a real app this can be a localization issue.

I also tend to like holding onto the actual exception object in a class like this, but opinions on that vary.